sdk/storage/Azure.Storage.Blobs/src/BlobDirectoryClient.cs

```diff
@@ -0,0 +1,726 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Cryptography;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+using Tags = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.Blobs.Specialized
+{
+    /// <summary>
+    /// Represents a virtual directory of blobs, designated by a delimiter character.
+    ///
+    /// Blob Directories cannot be created stand alone. In order to have a directory, there must be
+    /// at least one blob within the virtual directory in order for the directory to exist.
+    ///
+    /// The concept of non-virtual directories do not exist in the a flat namespace.
+    /// If you are looking to organize your data in a collection into a hierarchy of
+    /// directories and nested directories the same way a filesystem on your computer
+    /// is organized see
+    /// <see href="https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-namespace">
+    /// Azure Data Lake Storage Gen2 Hierarchical Namespace</see>.
+    /// </summary>
```

Comment:
It should just work for HNS, since it should just accept upload and download requests from the blobendpoint. Also since DataLakeDirectoryClient already exists, I think the only work we have to do there is add a UploadDirectory and DownloadDirectory API to the class. 

I will update the docs in a bit.